app/code/Magento/Backend/Setup/ConfigOptionsList.php

```diff
@@ -69,7 +69,7 @@ public function validate(array $options, DeploymentConfig $deploymentConfig)
     {
         $errors = [];
         if (isset($options[self::INPUT_KEY_BACKEND_FRONTNAME])
-            && !preg_match('/^[a-zA-Z0-9_]+$/', $options[self::INPUT_KEY_BACKEND_FRONTNAME])
+            && !preg_match('/^\w+$/', $options[self::INPUT_KEY_BACKEND_FRONTNAME])
```

Comment:
Such change is not needed, please revert all.

Reply:
@orlangur 

I don't agree keeping those regular expressions if there is an 'escape sequence' for that, in my mind is reinventing the wheel.

https://www.php.net/manual/en/regexp.reference.escape.php

Let me know again if you don't agree I'll close both pull request I made: this one and the back port https://github.com/magento/magento2/pull/22208

Reply:
Still not resolved. `[a-zA-Z0-9_]` is much more understandable. sometimes we need to add hyphen, for example.